hw/ip/otbn/util/shared/information_flow.py

```diff
@@ -0,0 +1,556 @@
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+
+from typing import Dict, List, Optional, Sequence, Set
+
+from serialize.parse_helpers import check_keys, check_list, check_str
+
+from .operand import Operand, RegOperandType
+
+FLAG_NAMES = ['c', 'm', 'l', 'z']
+SPECIAL_REG_NAMES = ['mod', 'acc']
+
+
+class InformationFlowGraph:
+    '''Represents an information flow graph.
+
+    The graph is represented as some set of "sink" nodes, each of which
+    recieves information from some set of "source" nodes. Any nodes which do
+    not appear in the "sink" list are expected to be unmodified; if a node is
+    cleared, for example set to a constant, it should appear as a sink with an
+    empty set of sources. Nodes are represented as strings.
+    '''
+    def __init__(self, flow: Dict[str, Set[str]]):
+        self.flow = flow
```

Comment:
It seems like flow has a somewhat complex semantic... Can you add a comment explaining what its structure is?

Reply:
Updated the class docstring to explain the structure, how is it now? (Sorry, the text that was there before was slightly confusing because I forgot to update it after making some changes.)